packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -70,6 +70,10 @@ export const isValidCodeWhispererCoreConnection = (conn?: Connection): conn is C
 }
 /** Superset that includes all of CodeWhisperer + Amazon Q */
 export const isValidAmazonQConnection = (conn?: Connection): conn is Connection => {
+    if (isSageMaker() && isAmazonQ()) {
+        return isIamConnection(conn) || (isSsoConnection(conn) && hasScopes(conn, amazonQScopes))
+    }
+
     return (
         (isSsoConnection(conn) || isBuilderIdConnection(conn)) &&
```

Comment:
Simplify
```suggestion
    return (
        (isAnySsoConnection(conn) || (isIamConnection(conn) && isSageMaker()) &&
```

Reply:
I extracted sagemaker condition into a separate function for better readability